clash-lib/src/Clash/Rewrite/Util.hs

```diff
@@ -829,9 +829,9 @@ specialise' specMapLbl specHistLbl specLimitLbl (TransformContext is0 _) e (Var
                       -- are inlined, meaning the state-transition-function
                       -- and the memory element will be in a single function.
                       gTmM <- fmap (lookupUniqMap (varName g)) $ Lens.use bindings
-                      return (g,maybe inl bindingSpec gTmM, maybe specArg (Left . (`mkApps` gArgs) . bindingTerm) gTmM)
-                    else return (f,inl,specArg)
-                _ -> return (f,inl,specArg)
+                      changed (g,maybe inl bindingSpec gTmM, maybe specArg (Left . (`mkApps` gArgs) . bindingTerm) gTmM)
+                    else changed (f,inl,specArg)
+                _ -> changed (f,inl,specArg)
```

Comment:
We already do a `changed` on https://github.com/clash-lang/clash-compiler/pull/1852/files#diff-ef03bca01fe87dd3c262b523212da81f7626bdc2a0f9af328fe0376bcc3f65b1L843 why would we need these?

Reply:
We might not need these, I should double check. The one on line 834 was just the first thing I found in common on all the tests where the "changed without notice" error was being thrown

Reply:
Yep, we don't need any of these